bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/EntsoeHandler.java

```diff
@@ -0,0 +1,253 @@
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(EntsoeConfiguration.class);
+
+        if (historicDaysInitially == 0) {
+            historicDaysInitially = config.historicDays;
+        }
+
+        if (isLinked(EntsoeBindingConstants.CHANNEL_SPOT_PRICE)) {
+            updateStatus(ThingStatus.UNKNOWN);
+        } else {
+            updateStatus(ThingStatus.ONLINE);
+        }
+
+        refreshJob = scheduler.schedule(this::refreshPrices, 0, TimeUnit.SECONDS);
+    }
+
+    private ZonedDateTime currentCetTime() {
+        return ZonedDateTime.now(cetZoneId);
+    }
+
+    private ZonedDateTime currentUtcTime() {
+        return ZonedDateTime.now(utcZoneId);
+    }
+
+    private ZonedDateTime currentUtcTimeWholeHours() {
+        return currentUtcTime().truncatedTo(ChronoUnit.HOURS);
+    }
+
+    private Optional<EntsoeTimeSerie> getCurrentHourEntsoeTimeSerie() throws EntsoeResponseMapException {
+        if (entsoeTimeSeries.isEmpty()) {
+            throw new EntsoeResponseMapException("responseMap is empty");
+        }
+        Instant currentHourStart = Instant.now().truncatedTo(ChronoUnit.HOURS);
+        if (!entsoeTimeSeries.containsKey(currentHourStart)) {
+            throw new EntsoeResponseMapException(
+                    String.format("Could not find a value for instant {} in responseMap", currentHourStart));
+        } else {
+            return entsoeTimeSeries.getOrDefault(currentHourStart, Optional.empty());
+        }
+    }
+
+    private boolean needToFetchHistoricDays() {
+        return needToFetchHistoricDays(false);
+    }
+
+    private boolean needToFetchHistoricDays(boolean updateHistoricDaysInitially) {
+        boolean needToFetch = false;
+        if (historicDaysInitially < config.historicDays) {
+            logger.debug("Need to fetch historic data. Historicdays was changed to a greater number: {}",
+                    config.historicDays);
+            needToFetch = true;
+        }
+
+        if (updateHistoricDaysInitially && historicDaysInitially != config.historicDays) {
+            historicDaysInitially = config.historicDays;
+        }
+
+        return needToFetch;
+    }
+
+    private void refreshPrices() {
+        if (!isLinked(EntsoeBindingConstants.CHANNEL_SPOT_PRICE)) {
+            logger.debug("Channel {} is not linked, skipping channel update",
+                    EntsoeBindingConstants.CHANNEL_SPOT_PRICE);
+            return;
+        }
+
+        config = getConfigAs(EntsoeConfiguration.class);
+
+        if (entsoeTimeSeries.isEmpty()) {
+            fetchNewPrices();
+            return;
+        }
+
+        boolean needsInitialUpdate = lastDayAheadReceived.equals(ZonedDateTime.of(LocalDateTime.MIN, cetZoneId))
+                || needToFetchHistoricDays(true);
+        boolean hasNextDayValue = entsoeTimeSeries
+                .containsKey(currentUtcTimeWholeHours().withHour(22).plusDays(1).toInstant());
```

Comment:
I'm wondering if this is correct. At least in Denmark we have prices until midnight tomorrow local time, so last hour start is 23:00. In UTC this is currently 21:00, but when DST is over, it will be 22:00. So I'm wondering if you need to use 23:00 in  CET and convert that to `Instant`?

Reply:
Yeah, so after summertime is over the UTC 22 will no longer equal 00 CET. So better to say 00 CET and convert into UTC, right?